File: sol/solution/0001-0100/0001. Two Sum/Solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    d = {}
    nums.each_with_index do |x, i|
        if d.key?(target - x)
            return [d[target - x], i]
        end
        d[x] = i
    end
end

File: sol/solution/0001-0100/0001. Two Sum/Solution.go
func twoSum(nums []int, target int) []int {
	d = map[int]int{}
	for i := 0; i < len(nums); i++ {
		if j, ok := d[target-nums[i]]; ok {
			return []int{j, i}
		}
		d[nums[i]] = i
	}
}